tests/machine.test.js

```diff
@@ -1134,4 +1134,177 @@ describe(OlmMachine.name, () => {
             expect(progressListener).toHaveBeenCalledWith(0, 1);
         });
     });
+
+    describe("Request missing secrets", () => {
+        async function getHypotheticalKeyQuery(new_user, new_device, bootstrapCrossSigning) {
+            let initialMachine = await OlmMachine.initialize(new_user, new_device);
+            const userId = initialMachine.userId.toString();
+            const deviceId = initialMachine.deviceId.toString();
+
+            let deviceKeys;
+            let outgoingRequests = await initialMachine.outgoingRequests();
+            for (const request of outgoingRequests) {
+                if (request instanceof KeysUploadRequest) {
+                    deviceKeys = JSON.parse(request.body);
+                }
+            }
+            delete deviceKeys.one_time_keys;
+            delete deviceKeys.fallback_keys;
+
+            let xsigning;
+
+            if (bootstrapCrossSigning) {
+                let bootstrapRequest = await initialMachine.bootstrapCrossSigning(true);
+
+                xsigning = JSON.parse(bootstrapRequest[0].body);
+                const newSignature = JSON.parse(bootstrapRequest[1].body);
+
+                console.log("Device keys", JSON.stringify(deviceKeys));
+                console.log("Xsigning", JSON.stringify(xsigning));
+                console.log("newSignature", JSON.stringify(newSignature));
```

Comment:
doesn't it log the right thing without `JSON.stringify`?

Also, this looks like it might be rather noisy?

Reply:
yes forgot to remove them